lib/breeds_list/widgets/breed_card.dart

```diff
@@ -37,15 +41,36 @@ class BreedCard extends StatelessWidget {
               children: [
                 Text('Origin: ${breed.origin}'),
                 if (breed.url != null)
-                  IconButton(
-                    onPressed: onPressed,
-                    icon: Icon(Icons.share),
+                  GestureDetector(
+                    child: IconButton(
+                      onPressed: onPressed,
+                      icon: Icon(Icons.photo_camera),
+                    ),
                   ),
-              ],
-            ),
+                Text('Add in'),
+                IconButton(
+                  onPressed: ()async {
+                    final controller = context.read<BreedsListController>();
+                    await controller.addFavouritesImages(breed.referenceImageId);
```

Comment:
can be simplified to 
context.read<BreedsListController>().addFavouritesImages(breed.referenceImageId);
without await